internal/server/server.go

```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/mainflux/mainflux/logger"
+)
+
+type Server interface {
+	Start() error
+	Stop() error
+}
+
+type BaseServer struct {
+	Ctx      context.Context
+	Cancel   context.CancelFunc
+	Name     string
+	Address  string
```

Comment:
Do we need `Address` field?

Reply:
@dborovcanin, Yes required. Now it is used properly.